### ISSUE TITLE
CMake not installed by install-build-deps.sh on Ubuntu 16.04

### ISSUE THREAD

**Issue by probonopd** [AppImage/AppImageKit#1007]
@knarfS reports:

> When building AppImageKit the `install-build-deps.sh` only checks for Ubuntu 14.04 to install a newer version of `cmake`, but shouldn't check it also for 16.04, because 14.04 LTS has expired in April 2019. I guess more and more people will use 16.04 instead? And what's about Ubuntu 32bit (you have to compile a newer `cmake` yourself)? 

To be honest, I never build locally, we have a quite involved setup using Docker and CentOS 6 for this repository to ensure increased binary compatibility with older systems. So yes, `install-build-deps.sh` should probably be updated.

> Shouldn't the build process work with the `cmake` version that is supplied by Ubuntu 14.04/16.04?

This is a question for @TheAssassin: Is the  `cmake` version that is supplied by Ubuntu 16.04 sufficient to build AppImageKit?

**Comment by TheAssassin** [comment]
No, we can not. There's features missing that we need and save a lot of code. Don't you think we tried to keep the compatibility as low as possible?

**Comment by TheAssassin** [comment]
This install script thingy should just be removed entirely. Such sutff is *never* maintained. Rather document which packages were needed in combination with a timestamp when this information was last updated.

**Comment by knarfS** [comment]
@TheAssassin Thanks for your reply. I was thinking, that there is a reason for your choice of the used cmake version.

I just found out, that Kitware offers a repository for Ubuntu 16.04/18.04 for cmake (https://apt.kitware.com/).That makes things a lot easier.

**Comment by TheAssassin** [comment]
Sorry for my slightly snippy reply beforehand.

As far as I remember, the reason for 3.6.0 is that it introduced `IMPORTED_TARGET` for pkg-config, which makes a lot of things easier. But there might be more. 

We try to keep these things as low as possible, but sometimes that's just not viable. I wasn't aware of that apt repository myself until now, thanks a lot for the link!

From time to time I build CMake on CentOS 6 on my CI and publish the artifacts here: https://artifacts.assassinate-you.net/artifactory/webapp/#/artifacts/browse/tree/General/cmake-centos6

You're free to use them, I will at some point even add ARM binaries.
